src/R2_alg.jl

```diff
@@ -162,7 +277,7 @@ function R2(
 
     # define model
     φk(d) = dot(∇fk, d)
-    mk(d) = φk(d) + ψ(d)
+    mk(d)::R = φk(d) + ψ(d)::R
```

Comment:
Still some allocations here but I could not fix it, maybe it is more related to ProximalOperators.